quipucords/utils/misc.py

```diff
@@ -34,3 +36,16 @@ def sanitize_for_utf8_compatibility(value):
     if isinstance(value, tuple):
         return tuple(sanitize_for_utf8_compatibility(value) for value in value)
     return value
+
+
+def file_exists(file_path: str | None) -> bool:
+    """Check if a file exists, has non-zero size, and is readable."""
+    if not file_path:
+        return False
+    path = Path(file_path).absolute()
+    return (
+        path.exists()
+        and path.is_file()
+        and path.stat().st_size > 0
```

Comment:
I'm not _really_ worried, but what if file is 0 size and not writable?

That check will return False, and I assume we will try to write to file. But with writable bit turned off, we will fail. Not sure if cache generating code handles that.

Reply:
That's a good point about the writable bit. I've added that to the check here, but ultimately I consider handling "failure to write" to be outside the scope of this issue. Failing to write to the file will certainly cause Celery to blow up exceptions elsewhere that should be very obvious to the reader.